src/PerformanceSensitiveAnalyzers/CSharp/TypeConversionAllocationAnalyzer.cs

```diff
@@ -325,12 +325,22 @@ private static void CheckDelegateCreation(SyntaxNode node, TypeInfo typeInfo, Se
                     }
                 }
 
-                var symbolInfo = semanticModel.GetSymbolInfo(node, cancellationToken).Symbol;
-                if (symbolInfo?.ContainingType?.IsValueType == true && !insideObjectCreation)
+                if (IsStructInstanceMethod(node!, semanticModel, cancellationToken))
                 {
                     reportDiagnostic(Diagnostic.Create(DelegateOnStructInstanceRule, location, EmptyMessageArgs));
                 }
             }
         }
+
+        private static bool IsStructInstanceMethod(SyntaxNode? node, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (node.IsKind(SyntaxKind.AnonymousMethodExpression) || node.IsKind(SyntaxKind.ParenthesizedLambdaExpression))
```

Comment:
❓ Is this still correct if the lambda captures `this`?

Reply:
That would be a boxing by itself, right? I think this check is only for wrapping an instance method on struct

Reply:
It turns out it's not possible to box from within a lambda in a struct, so we're good there.